packages/fiber/src/core/reconciler.ts

```diff
@@ -219,7 +219,7 @@ function removeChild(
     if (typeof dispose === 'function') {
       scheduleCallback(idlePriority, () => {
         try {
-          dispose()
+          dispose.call(child.object)
```

Comment:
`child.object` can be mutated the time this runs. I'd prefer to `const dispose = child.object.dispose.bind(child.object)`.

Reply:
Okay, I think to use `bind` we need to check that we are binding to a function that actually exists first, otherwise it is fine.